lib/pure/httpclient.nim

```diff
@@ -382,7 +382,8 @@ proc format(p: MultipartData): tuple[header, body: string] =
 
 proc request*(url: string, httpMethod: string, extraHeaders = "",
               body = "", sslContext = defaultSSLContext, timeout = -1,
-              userAgent = defUserAgent, proxy: Proxy = nil): Response =
+              userAgent = defUserAgent, proxy: Proxy = nil,
+              domain: int = -1): Response =
```

Comment:
The user of this proc shouldn't have to specify the ``domain``. Both IPv4 and IPv6 should indeed be checked.